Source/Scene/Camera.js

```diff
@@ -1527,59 +1527,164 @@ define([
         });
     };
 
+    var scratchLookAtMatrix4 = new Matrix4();
+    var scratchLookAtTransformMatrix4 = new Matrix4();
+
     /**
-     * Sets the camera position and orientation with an eye position, target, and up vector.
-     * This method is not supported in 2D mode because there is only one direction to look.
+     * Sets the camera position and orientation using a target, offset and transformation matrix. The target must be given in
+     * world coordinates. The offset is a cartesian offset from the target in a reference frame about the target. By default,
+     * the reference frame is the local east-north-up frame centered at the target. The transformation matrix can be used to
+     * specify a different reference frame.
+     *
+     * In 2D, there must be a top down view. The camera will be placed above the target looking down. The height above the
+     * target will be the magnitude of the offset. The heading will be determined from the offset. If the heading cannot be
+     * determined from the offset, the heading will be north.
+     *
+     * @param {Object} options An object with the following properties:
+     * @param {Cartesian3} options.target The target position in world coordinates.
+     * @param {Cartesian3} options.offset The offset from the target in a reference frame centered at the target.
+     * @param {Matrix4} [options.transform] The transformation matrix defining the reference frame centered at the target. The default is a local east-north-up reference frame.
```

Comment:
When `transform` is defined, is `target` ignored?  It seems like this is ambiguous since the translation is basically provided in separate places.  It almost feels like this should be a 3x3 matrix or we should allow `(target && offset)` or `(offset && transform)` perhaps even using two separate functions.